nextcord/cache.py

```diff
@@ -64,46 +79,57 @@ class NullCache(BaseCache):
 
 class MemoryCache(BaseCache):
     _guilds: Dict[int, GuildData]
+    _members: Dict[Tuple[int, int], MemberData]
+    """{(guild_id, member_id): MemberData}"""
     _messages: dict[int, MessageData]
     _users: Dict[int, UserData]
 
-    def __init__(self, **kwargs):
-        super().__init__()
+    # noinspection PyProtocol
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
         self._guilds = {}
+        self._members = {}
         self._messages = {}
         self._users = {}
 
-    async def add_guild(self, guild_data: GuildData):
+    async def add_guild(self, guild_data: GuildData) -> None:
         self._guilds[int(guild_data["id"])] = guild_data
 
     async def get_guild(self, guild_id: int) -> Optional[GuildData]:
         return self._guilds.get(guild_id, None)
 
-    async def get_guild_ids(self) -> Iterable[int]:
+    async def get_guild_ids(self) -> Sequence[int]:
         return tuple(self._guilds.keys())
 
-    async def add_message(self, message_data: MessageData):
+    async def add_member(self, member_data: MemberData, guild_id: int) -> None:
+        self._members[(int(member_data["user"]["id"]), guild_id)] = member_data
+
+    async def remove_member(self, member_id: int, guild_id: int) -> bool:
+        return self._members.pop((member_id, guild_id)) is not None
```

Comment:
`dict.pop` actually raises `KeyError` if the key provided isn't found in the dictionary.